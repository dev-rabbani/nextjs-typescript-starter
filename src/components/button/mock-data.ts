import {
  FillButtonProps,
  FillRedirectButtonProps,
  OutlineButtonProps,
  OutlineRedirectButtonProps,
} from './interface';

export const buttonProps: FillButtonProps = {
  label: 'Button',
  colorScheme: 'bg-primary-text-white',
  hoverColorScheme: 'bg-danger-text-white',
  size: 'lg',
  clickHandler: () => {},
  variant: 'fill',
};

export const redirectButtonProps: FillRedirectButtonProps = {
  href: 'https://www.google.com/',
  isOpenNewTab: true,
  label: 'Redirect Button',
  colorScheme: 'bg-primary-text-white',
  hoverColorScheme: 'bg-danger-text-white',
  size: 'lg',
  variant: 'fill',
};

export const outlineButtonProps: OutlineButtonProps = {
  label: 'Outline Button',
  colorScheme: 'bg-transparent-text-primary',
  hoverColorScheme: 'bg-transparent-text-warning',
  size: 'lg',
  clickHandler: () => {},
  variant: 'outline',
};

export const redirectOutlineButtonProps: OutlineRedirectButtonProps = {
  href: 'https://www.google.com/',
  isOpenNewTab: true,
  label: 'Outline Redirect Button',
  colorScheme: 'bg-transparent-text-primary',
  hoverColorScheme: 'bg-transparent-text-warning',
  size: 'lg',
  variant: 'outline',
};
