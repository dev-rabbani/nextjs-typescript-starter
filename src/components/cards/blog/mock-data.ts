import { BlogCardProps } from './interface';

export const blogCardProps: BlogCardProps = {
  slug: 'https://example.com/',
  image: {
    src: 'https://res.cloudinary.com/dktw59lc2/image/upload/v1669267967/BrightonSEO/Carousel/image-2_yph8q6.jpg',
    alt: 'dummy image 2',
  },
  title: 'Headless Website Development',
};
