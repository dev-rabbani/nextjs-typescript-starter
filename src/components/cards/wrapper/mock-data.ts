import { CardWrapperProps } from './interface';

export const cardWrapperProps: CardWrapperProps = {
  cardBorderRadius: 'radius-none',
  cardShadow: 'shadow-one',
};
