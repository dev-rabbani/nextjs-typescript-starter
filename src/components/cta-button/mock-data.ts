import { ICtaButton } from './interface';

export const ctaButtonProps: ICtaButton = {
  label: 'Call to action',
  href: '/',
  colorScheme: 'primary',
  isOpenNewTab: false,
};
