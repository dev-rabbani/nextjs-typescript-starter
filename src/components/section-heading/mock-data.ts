import { ISectionHeading } from './interface';

export const sectionHeadingProps: ISectionHeading = {
  subtitle: 'Sub Title',
  title: 'What Makes Us Different?',
  description:
    'Lorem ipsum dolor, sit amet consectetur adipisicing elit. Dolorem est fugiat ea voluptatem rerum unde necessitatibus, eos ex maxime laborum?',
  colorScheme: 'primary',
  hasBottomSpacing: true,
  isCenter: false,
};
