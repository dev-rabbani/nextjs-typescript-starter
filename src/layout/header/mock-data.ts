import { IHeader } from './interface';

export const headerProps: IHeader = {
  logo: 'Starter',
  menuItems: [
    {
      label: 'Home',
      href: '/',
    },
    {
      label: 'About',
      href: '/',
    },
    {
      label: 'Service',
      href: '/',
    },
    {
      label: 'Blog',
      href: '/',
    },
  ],
};
