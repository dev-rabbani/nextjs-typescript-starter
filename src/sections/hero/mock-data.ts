import { IHero } from './interface';

export const heroProps: IHero = {
  title:
    'Welcome to Nextjs typescript starter, the ultimate solution for building modern web applications with Next.js!',
  description:
    'Our starter template is designed to give you a head start on your project, with all the tools you need to create a beautiful and responsive web application in no time. With Next.js, you can enjoy the benefits of server-side rendering and a streamlined development process, while still having the flexibility to use the front-end framework of your choice',
  buttons: {
    buttonLeft: {
      label: 'Server Side Example',
      href: '/',
    },
    buttonRight: {
      label: 'Client Side Example',
      href: 'client-side',
    },
  },
};
