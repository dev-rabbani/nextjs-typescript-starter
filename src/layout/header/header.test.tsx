import { render } from '@testing-library/react';
import Header from '.';
import { headerProps } from './mock-data';

it('render Header correctly', () => {
  const { getAllByRole } = render(<Header {...headerProps} />);
  const element = getAllByRole('presentation')[0];

  expect(element).toBeInTheDocument();
  expect(element).toMatchSnapshot();
});
