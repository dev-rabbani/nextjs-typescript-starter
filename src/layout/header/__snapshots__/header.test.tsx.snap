// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Header correctly 1`] = `
<header
  class="bg-black"
  role="presentation"
>
  <div
    class="container"
    role="presentation"
  >
    <div
      class="flex flex-col lg:flex-row lg:items-center lg:justify-between"
    >
      <h1
        class="text-4xl text-white"
      >
        <a
          href="/"
        >
          Starter
        </a>
      </h1>
      <nav>
        <ul
          class="flex flex-col gap-5 lg:flex-row lg:items-center lg:gap-12"
        >
          <li>
            <a
              class="inline-flex items-end py-[38px] text-base font-bold text-white transition hover:text-primary"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              class="inline-flex items-end py-[38px] text-base font-bold text-white transition hover:text-primary"
              href="/"
            >
              About
            </a>
          </li>
          <li>
            <a
              class="inline-flex items-end py-[38px] text-base font-bold text-white transition hover:text-primary"
              href="/"
            >
              Service
            </a>
          </li>
          <li>
            <a
              class="inline-flex items-end py-[38px] text-base font-bold text-white transition hover:text-primary"
              href="/"
            >
              Blog
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </div>
</header>
`;
